iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -237,11 +238,18 @@ public static void rollback(Table table, AlterTableExecuteSpec.RollbackSpec.Roll
    * @param table the iceberg table
    * @param value parameter of the rollback, that can be a timestamp in millis or a snapshot id
    */
-  public static void setCurrentSnapshot(Table table, Long value) {
+  public static void setCurrentSnapshot(Table table, String value) {
     ManageSnapshots manageSnapshots = table.manageSnapshots();
+    long snapshotId;
+    try {
+      snapshotId = Long.valueOf(value);
+    } catch (NumberFormatException e) {
+      snapshotId = Optional.ofNullable(table.refs().get(value)).map(ref -> ref.snapshotId()).orElseThrow(() ->
+          new IllegalArgumentException(String.format("SnapshotRef %s does not exist", value)));
+    }
     LOG.debug("Rolling the iceberg table {} from snapshot id {} to snapshot ID {}", table.name(),
```

Comment:
can you update the log and javadoc accordingly that we now support ID+ref

Reply:
Fixed.